digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/linearasset/PostGISSpeedLimitDao.scala

```diff
@@ -61,6 +98,129 @@ class PostGISSpeedLimitDao(val roadLinkService: RoadLinkService) extends Dynamic
     }
   }
 
+  def fetchByBBoxExperimental(bbox: BoundingRectangle): (Seq[PieceWiseLinearAsset], Seq[RoadLinkForUnknownGeneration]) = {
+    val bboxFilter = PostGISDatabase.boundingBoxFilter(bbox, "shape")
+
+    val query =
+      sql"""
+      WITH cte_kgv_roadlink AS (
+        SELECT
+          kgv.linkid,
+          kgv.directiontype,
+          kgv.shape,
+          kgv.geometrylength,
+          kgv.adminclass,
+          kgv.municipalitycode,
+          kgv.constructiontype
+        FROM kgv_roadlink kgv
+        JOIN link_type lt ON kgv.linkid = lt.link_id
+        JOIN functional_class fc ON kgv.linkid = fc.link_id
+        WHERE #$bboxFilter
+          AND kgv.expired_date IS NULL
+          AND kgv.constructiontype NOT IN (1, 2)
+          AND kgv.mtkclass NOT IN (12318, 12312) -- Filter out HardShoulder and WinterRoad links
+          AND (kgv.adminclass != 1 OR (kgv.adminclass = 1 AND lt.link_type NOT IN (7, 8, 9, 10, 12, 13, 14, 15, 21))) -- Filter out PedestrianZone and CableFerry links
```

Comment:
Ei oo ehkä kaikkein helppolukuisin, kun on näin monta ehtoa sql:ssä. Homman nimi tais olla kuitenkin nopeuttaminen eli, jos auttaa siihen niin ok. Kiva, kun on kommentit.